### PR TITLE
OpenSora 1.1 dynamic shape training

### PR DESCRIPTION
**Pull request by zhtmike**
Minimum change for Dongli's branch:  https://github.com/yuedongli1/mindone/tree/jitO0_dynamic_shape

# What does this PR do?

Fixes # (issue)

Adds # (feature)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/mindspore-lab/mindone/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the documentation with your changes? E.g. record bug fixes or new features in `What's New`. Here are the
      [documentation guidelines](https://github.com/mindspore-lab/mindcv/wiki/%E6%96%87%E6%A1%A3%E7%BC%96%E5%86%99%E8%A7%84%E8%8C%83)
- [ ] Did you build and run the code without any errors?
- [ ] Did you report the running environment (NPU type/MS version) and performance in the doc? (better record it for data loading, model inference, or training tasks)
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@xxx

### CODE REVIEW

File: examples/opensora_hpcai/scripts/train.py [hunk 1]
```diff
@@ -605,5 +616,6 @@ def main(args):
 
 if __name__ == "__main__":
     logger.debug("process id:", os.getpid())
+    ms.set_context(graph_kernel_flags='--disable_packet_ops=Reshape')
```
Comment (hadipash) [inline examples/opensora_hpcai/scripts/train.py:619]: Move it inside `init_env`.

File: examples/opensora_hpcai/opensora/models/layers/blocks.py [hunk 2]
```diff
@@ -490,7 +491,7 @@ def __init__(self, patch_size: int = 2, in_chans: int = 3, embed_dim: int = 96,
         self.patch_size: Tuple = (patch_size, patch_size) if isinstance(patch_size, int) else patch_size
         self.embed_dim = embed_dim
         self.proj = nn.Conv2d(
-            in_chans, embed_dim, kernel_size=patch_size, stride=patch_size, pad_mode="same", has_bias=bias
+            in_chans, embed_dim, kernel_size=patch_size, stride=patch_size, pad_mode="valid", has_bias=bias
```
[the remark this answers is not in the document]
Comment (zhtmike) [reply examples/opensora_hpcai/opensora/models/layers/blocks.py:494]: Agree, it is not work with [640, 360] shape inference now

File: examples/opensora_hpcai/opensora/models/layers/blocks.py [hunk 3]
```diff
@@ -466,8 +469,6 @@ def token_drop(self, caption, force_drop_ids=None):
         return caption
 
     def construct(self, caption, train, force_drop_ids=None):
-        if train:
-            assert caption.shape[2:] == self.y_embedding.shape
```
[the remark this answers is not in the document]
Comment (zhtmike) [reply examples/opensora_hpcai/opensora/models/layers/blocks.py:470]: I guess now the dynamic shape does not behave well with if else branch

**Comment by SamitHuang** [comment]
After the set_inputs changes, how is the influence on training performance on fixed shape video like 576x1024x24?

File: examples/opensora_hpcai/scripts/train.py (hunk 1 above)
Comment (zhtmike) [reply examples/opensora_hpcai/scripts/train.py]: > Move it inside `init_env`.

done

**Comment by zhtmike** [comment]
> After the set_inputs changes, how is the influence on training performance on fixed shape video like 576x1024x24?

it will definitely slow down the training time. Now it will be settled only if it is necessary

File: examples/opensora_hpcai/opensora/datasets/video_dataset_refactored.py [hunk 4]
```diff
@@ -83,12 +83,12 @@ def __init__(
 
         self.output_columns = output_columns
         if self._buckets is not None:
-            assert vae_latent_folder is None, "`vae_latent_folder` is not supported with bucketing"
+            # assert vae_latent_folder is None, "`vae_latent_folder` is not supported with bucketing"
```
Comment (CaitinZhao) [inline examples/opensora_hpcai/opensora/datasets/video_dataset_refactored.py:86]: 这个不用改

File: examples/opensora_hpcai/opensora/models/layers/blocks.py [hunk 5]
```diff
@@ -102,7 +102,7 @@ class MultiHeadCrossAttention(nn.Cell):
 
     def __init__(self, d_model, num_heads, attn_drop=0.0, proj_drop=0.0, has_bias=True, enable_flash_attention=False):
         super().__init__()
-        assert d_model % num_heads == 0, "d_model must be divisible by num_heads"
+        # assert d_model % num_heads == 0, "d_model must be divisible by num_heads"
```
Comment (CaitinZhao) [inline examples/opensora_hpcai/opensora/models/layers/blocks.py:105]: ditto

File: examples/opensora_hpcai/opensora/models/layers/blocks.py [hunk 6]
```diff
@@ -117,7 +117,7 @@ def __init__(self, d_model, num_heads, attn_drop=0.0, proj_drop=0.0, has_bias=Tr
         )
         if self.enable_flash_attention:
             attn_dtype = ms.bfloat16
-            assert attn_drop == 0.0, "attn drop is not supported in FA currently."
+            # assert attn_drop == 0.0, "attn drop is not supported in FA currently."
```
Comment (CaitinZhao) [inline examples/opensora_hpcai/opensora/models/layers/blocks.py:120]: ditto

File: examples/opensora_hpcai/opensora/models/layers/blocks.py [hunk 7]
```diff
@@ -215,7 +216,7 @@ def __init__(
         rope=None,
     ):
         super().__init__()
-        assert dim % num_heads == 0, "dim should be divisible by num_heads"
+        # assert dim % num_heads == 0, "dim should be divisible by num_heads"
```
Comment (CaitinZhao) [inline examples/opensora_hpcai/opensora/models/layers/blocks.py:219]: ditto

File: examples/opensora_hpcai/opensora/datasets/video_dataset_refactored.py [hunk 8]
```diff
@@ -83,12 +83,12 @@ def __init__(
...
             self.output_columns += ["bucket_id"]  # pass bucket id information to transformations
 
         if self._pre_patchify:
             self._patch_size = patch_size
-            assert self._patch_size[0] == 1
+            # assert self._patch_size[0] == 1
```
Comment (CaitinZhao) [inline examples/opensora_hpcai/opensora/datasets/video_dataset_refactored.py:91]: ditto

File: examples/opensora_hpcai/opensora/models/layers/blocks.py [hunk 9]
```diff
@@ -416,13 +417,15 @@ def construct(
...
+        if T is None:
+            T = self.d_t
+        if S is None:
+            S = self.d_s
+        shift, scale = mint.chunk(self.scale_shift_table[None] + t[:, None], 2, 1)
         x = t2i_modulate(self.norm_final(x), shift, scale)
 
-        if frames_mask is not None:
-            shift_zero, scale_zero = (self.scale_shift_table[None] + t0[:, None]).chunk(2, axis=1)
+        if True:
```
Comment (CaitinZhao) [inline examples/opensora_hpcai/opensora/models/layers/blocks.py:427]:
```suggestion
        if self.has_frames_mask:
```
add `self.has_frames_mask = True`  in __init__ when v1.0

File: examples/opensora_hpcai/opensora/models/layers/blocks.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (CaitinZhao) [reply examples/opensora_hpcai/opensora/models/layers/blocks.py:494]: calculate a pad, and set pad mode？

File: examples/opensora_hpcai/opensora/models/layers/blocks.py [hunk 10]
```diff
@@ -687,7 +686,7 @@ class PositionEmbedding2D(nn.Cell):
     def __init__(self, dim: int):
         super().__init__()
         self.dim = dim
-        assert dim % 4 == 0, "dim must be divisible by 4"
+        # assert dim % 4 == 0, "dim must be divisible by 4"
```
Comment (CaitinZhao) [inline examples/opensora_hpcai/opensora/models/layers/blocks.py:689]: no need delete

File: examples/opensora_hpcai/opensora/models/stdit/stdit2.py [hunk 11]
```diff
@@ -46,7 +46,7 @@ def __init__(
         self.hidden_size = hidden_size
         self.enable_flashattn = enable_flashattn
 
-        assert not enable_layernorm_kernel, "Not implemented"
+        # assert not enable_layernorm_kernel, "Not implemented"
```
Comment (CaitinZhao) [inline examples/opensora_hpcai/opensora/models/stdit/stdit2.py:49]: ditto

File: examples/opensora_hpcai/opensora/models/stdit/stdit2.py [hunk 12]
```diff
@@ -112,37 +112,37 @@ def construct(
...
         shift_msa_zero, scale_msa_zero, gate_msa_zero, shift_mlp_zero, scale_mlp_zero, gate_mlp_zero = (None,) * 6
         shift_tmp_zero, scale_tmp_zero, gate_tmp_zero = (None,) * 3
-        if frames_mask is not None:
-            shift_msa_zero, scale_msa_zero, gate_msa_zero, shift_mlp_zero, scale_mlp_zero, gate_mlp_zero = (
-                self.scale_shift_table[None] + t0.reshape(B, 6, -1)
-            ).chunk(6, axis=1)
-            shift_tmp_zero, scale_tmp_zero, gate_tmp_zero = (
-                self.scale_shift_table_temporal[None] + t0_tmp.reshape(B, 3, -1)
-            ).chunk(3, axis=1)
+        if True:
```
Comment (CaitinZhao) [inline examples/opensora_hpcai/opensora/models/stdit/stdit2.py:124]: ditto

**Comment by zhtmike** [comment]
move to #605

**Pull request closed without merging**